test/src/test/kotlin/graphql/nadel/tests/EngineTests.kt

```diff
@@ -180,7 +189,79 @@ private suspend fun execute(
 
                                 if (indexOfCall != null) {
                                     val serviceCall = serviceCalls.removeAt(indexOfCall)
-                                    serviceCall.response
+                                    if (serviceCall.incrementalResponse != null && serviceCall.response != null) {
+                                        fail(
+                                            """Cannot have both an incremental and non-incremental response 
+                                        |   fixture : '${fixture.name}' 
+                                        |   service : '${serviceName}' 
+                                        |   service : '${serviceName}' 
+                                        |   query : '${actualQuery}' 
+                                        |   variables : '${actualVariables}' 
+                                        |   operation : '${actualOperationName}' 
+                                        """.trimMargin()
+                                        )
+                                    }
+                                    else if (serviceCall.incrementalResponse != null) {
+                                        fun transformData(executionResult: JsonMap): DelayedIncrementalPartialResult{
+                                            val incrementalDataVal = executionResult["incremental"] as List<JsonMap> //[0]["data"]
+                                            return newIncrementalExecutionResult()
+                                                .hasNext(executionResult["hasNext"] as Boolean)
+                                                .apply {
+                                                    if(executionResult["extensions"] != null) extensions(executionResult["extensions"] as Map<Any, Any>)
+                                                }
+                                                .incrementalItems(
+                                                    incrementalDataVal.map{
+                                                        DeferPayload.newDeferredItem()
+                                                            .label(it["label"] as String)
+                                                            .path(it["path"] as List<Object>)
+                                                            .apply {
+                                                                if (it["extensions"] != null) extensions(it["extensions"] as Map<Any, Any>)
+                                                                if (it["errors"] != null) errors(it["errors"] as List<GraphQLError>)
+                                                            }
```

Comment:
only label and path are required as per spec here:
https://github.com/graphql/graphql-spec/blob/c630301560d9819d33255d3ba00f548e8abbcdc6/spec/Section%207%20--%20Response.md?plain=1#L376C19-L377C53